### ISSUE TITLE
The cmdlet cannot run because the -ContentType parameter is not a valid Content-Type header.

### ISSUE THREAD

**Issue by Clifra-Jones** [PowerShell/PowerShell#17855]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I have a Rest API Call that is supposed to upload a file.
I keep getting this error
The cmdlet cannot run because the -ContentType parameter is not a valid Content-Type header. Specify a valid Content-Type for -ContentType, then retry.

My header looks like this
`{
  "Content-Disposition": "attachment; filename=Shadow Copy Restore.docx",
  "Authorization": "Bearer ....",
  "Content-Length": "FILE_SIZE",
  "Content-Type": "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
}`

I've tried the content-disposition with and without quotes around the file name.

I've created the body of the request with:
`$body = [System.IO.File]::ReadAllBytes($path)`

Unfortunately the -InFile parameter does not seen to work with this API.
I have also tried a content type of application/octet-stream and get the same results.

### Expected behavior

```console
No error
```

### Actual behavior

```console
The cmdlet cannot run because the -ContentType parameter is not a valid Content-Type header. Specify a valid Content-Type for -ContentType, then retry.
```

### Error details

```console
The cmdlet cannot run because the -ContentType parameter is not a valid Content-Type header. Specify a valid Content-Type for -ContentType, then retry.
To suppress header validation, supply the -SkipHeaderValidation parameter.
```

### Environment data

```powershell
ame                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Linux 5.18.13-100.fc35.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Jul 22 14:20:24 UTC 2022
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by fMichaleczek** [comment]
Have you try "-SkipHeaderValidation" ? 

Nobody can reproduce your problem (and you forget to specify which command), you should provide a curl equivalent to show it's not a problem with the webserver.

If you want general support about PowerShell, StackOverflow or discussions is a better place to discuss.

**Comment by fMichaleczek** [comment]
@Clifra-Jones You should read this issue https://github.com/PowerShell/PowerShell/issues/2895#issuecomment-360118454 and the link to the blog.

I will create an issue in PowerShell Docs about "Header Validation"

**Comment by jhoneill** [comment]
Could you show a little code where you build the parameters and call Invoke-RestMethod / Invoke-WebRequest ? Generally, building a header like that isn't needed. The `-ContentType` is a command line parameter, not passed as header, the size is calculated for you 
I'd expect something like this to work 
```
$headers = @{authorization='bearer xyz123' }
invoke-restmethod -method Post -headers $headers -infile <whatever> -contentType <mimetime>
```

**Comment by Clifra-Jones** [comment]
> Could you show a little code where you build the parameters and call Invoke-RestMethod / Invoke-WebRequest ? Generally, building a header like that isn't needed. The `-ContentType` is a command line parameter, not passed as header, the size is calculated for you I'd expect something like this to work
> 
> ```
> $headers = @{authorization='bearer xyz123' }
> invoke-restmethod -method Post -headers $headers -infile <whatever> -contentType <mimetime>
> ```

Thanks @jhoneill I was able to get past the error with this.

**Comment by Clifra-Jones** [comment]
I closed this earlier today but I'm posting this because I solved this issue so that anyone looking this up can find the answer.
Using the -InFile parameter DOES NOT WORK with the Smartsheet API. After using POsyman and lookin at Postman's Powershell Code example I found the solution.

Manually build the headers as a hash table:
`Name                           Value
----                           -----
Content-Type                   application/vnd.openxmlformats-officedocument.wordprocessingml.document
Authorization                  Bearer VcjJj965oRo3E30PG1sXKkkvAFfFJhEhg5paj
Content-Disposition            attachment; filename="CWilliams.docx"`

Read the file in as a byte array:
`$body = [System.IO.File]::ReadAllBytes($path)`

Issue the following Command:
`$response = Invoke-RestMethod -Method 'POST' -Uri $Uri -Headers $Headers -Body $body`

This works.
I was trying this before but adding 'Content-Length ' to the Headers. If you add that it fails with odd errors.

What ever headers -InFile is producing doesn't work properly with the Smartsheet API.
